imageio/plugins/ffmpeg.py

```diff
@@ -471,24 +514,27 @@ def _open(self, fps=10, codec='libx264', bitrate=400000):
             # Initialize parameters
             self._proc = None
             self._size = None
-        
-        def _close(self):
-            # Close subprocess
-            if self._proc is not None:
-                self._proc.stdin.close()
-                self._proc.stderr.close()
-                self._proc.wait()
-                self._proc = None
-        
+            self._cmd = None
+
+        def _close(self, timeout=1.0):
+            if self._proc is None:  # pragma: no cover
+                return  # no process
+            if self._proc.poll() is not None:
+                return  # process already dead
+            # End process
+            # self._proc.stdin.close() # .communicate will take care of this
+            # The .communicate is preferred over .wait which can deadlock
+            self._proc.communicate()
```

Comment:
I don't think this is right. Are you sure it closes stdin (it does not say so in the docs)? If it does, writing should work, I suppose, but for *reading*, we really need to ask ffmpeg to terminate. I'm also slightly worried that if ffmpeg hangs, this hangs as well, but we can try and see if that's really a problem...